asab/library/providers/filesystem.py

```diff
@@ -24,29 +24,26 @@ async def read(self, path):
 			return None
 
 
-	async def list(self, path):
-		recursive = path.endswith("*")
+	async def list(self, path, recursive=True):
 
 		if recursive:
-			path = os.path.join(path[:-2], "**")
+			path = os.path.join(path, "**")
 
-		file_names = glob.iglob(
-			os.path.join(self.LibraryBaseDir, path, "*.yaml"),
-			recursive=recursive
+		file_names = list(
+			glob.iglob(
+				os.path.join(self.LibraryBaseDir, path),
+				recursive=recursive
+			)
 		)
 
-		files_names_list = []
-
 		# Remove library path from the beginning of file names
 		library_path_to_replace = "{}/".format(os.path.abspath(self.LibraryBaseDir))
-
-		for file_name in file_names:
-			files_names_list.append(
-				file_name.replace(library_path_to_replace, '')
-			)
+		for name in file_names:
+			assert name.startswith(library_path_to_replace)
+		file_names_list = [name[len(library_path_to_replace):] for name in file_names]
```

Comment:
I don't think it can really happen - that the name would start with something else... I don't know what error message I would write there, as I cannot imagine the situation. Anyway, when it happens, I find AssertionError easier to debug than some Error from list comprehension...